src/StyleableBannerNewsletter/index.tsx

```diff
@@ -0,0 +1,299 @@
+import React, { useState } from 'react';
+
+import {
+    NewsletterCtaButton,
+    defaultNewsletterCtaButtonColors,
+} from '../components/NewsletterCtaButton';
+import {
+    NewsletterFrequencyBlock,
+    defaultNewsletterFrequencyColors,
+} from '../components/NewsletterFrequencyBlock';
+import { BannerCloseButton, defaultBannerCloseButtonColors } from '../components/BannerCloseButton';
+
+import type { ColorValueHex } from '../logic/types';
+import type { NewsletterSubscribeCallback } from '../types/dcrTypes';
+import type { TrackClick } from '../utils/tracking';
+import type { BannerNewsletterBaseColorStyles } from '../styles/colorData';
+
+import { canRender, COMPONENT_NAME } from './canRender';
+export { COMPONENT_NAME };
+
+export type BrazeMessageProps = {
+    ophanComponentId?: string;
+    header?: string;
+    frequency?: string;
+    body?: string;
+    boldText?: string;
+    secondParagraph?: string;
+    newsletterCta?: string;
+    imageUrl?: string;
+    newsletterId?: string;
+
+    styleBackground?: ColorValueHex;
+    styleBody?: ColorValueHex;
+    styleClockColor?: ColorValueHex;
+    styleClose?: ColorValueHex;
+    styleCloseBackground?: ColorValueHex;
+    styleCloseHover?: ColorValueHex;
+    styleFrequencyText?: ColorValueHex;
+    styleHeader?: ColorValueHex;
+    styleBoldText?: ColorValueHex;
+    styleBoldTextBackground?: ColorValueHex;
+    styleNewsletterButton?: ColorValueHex;
+    styleNewsletterButtonBackground?: ColorValueHex;
+    styleNewsletterButtonHover?: ColorValueHex;
+    styleReminderAnimation?: ColorValueHex;
+    styleSecondParagraph?: ColorValueHex;
+};
+
+export type Props = {
+    brazeMessageProps: BrazeMessageProps;
+    trackClick: TrackClick;
+    subscribeToNewsletter?: NewsletterSubscribeCallback;
+};
+
+const StyleableBannerNewsletter: React.FC<Props> = (props: Props) => {
+    if (!props.subscribeToNewsletter || !canRender(props.brazeMessageProps)) {
+        return null;
+    }
+
+    const {
+        brazeMessageProps: {
+            ophanComponentId,
+            header,
+            body,
+            boldText,
+            secondParagraph,
+            newsletterCta = 'Sign up',
+            newsletterId,
+            imageUrl,
+            frequency,
+        },
+        subscribeToNewsletter,
+        trackClick,
+    } = props;
+
+    const brazeProps = props.brazeMessageProps;
+
+    const newsletterBaseColors = getColors(brazeProps, defaultBannerNewsletterBaseColors);
+    const newsletterCtaColors = getColors(brazeProps, defaultNewsletterCtaButtonColors);
+    const newsletterFrequencyColors = getColors(brazeProps, defaultNewsletterFrequencyColors);
+    const bannerCloseButtonColors = getColors(brazeProps, defaultBannerCloseButtonColors);
+
+    const styles = getBannerNewsletterStyles(brazeProps, newsletterBaseColors);
+
+    const [showBanner, setShowBanner] = useState(true);
+
+    if (!showBanner) {
+        return null;
+    }
+
+    console.log(styles);
```

Comment:
Should probably remove this console.log!

Reply:
Removed